packages/core/src/lib/base_protocol.ts

```diff
@@ -50,6 +50,56 @@ export class BaseProtocol implements IBaseProtocol {
     );
     return peer;
   }
+
+  /**
+   * Retrieves a list of peers based on the specified criteria.
+   *
+   * @param numPeers - The number of peers to retrieve. If 0, all peers are returned.
+   * @param includeBootstrap - If true, includes a bootstrap peer in the result. Useful for protocols like Filter and Store that require only one peer for now.
+   * @returns A Promise that resolves to an array of peers based on the specified criteria.
+   */
+  protected async getPeers({
+    numPeers,
+    includeBootstrap
+  }: {
+    numPeers: number;
+    includeBootstrap: boolean;
+  }): Promise<Peer[]> {
+    // Retrieve all peers that support the protocol
+    const allPeersForProtocol = await getPeersForProtocol(this.peerStore, [
+      this.multicodec
+    ]);
+
+    if (numPeers === 0) {
```

Comment:
probably filter on bootstrap if needed. As a consumer, I'd expect `getPeers({numPeers: 0, includeBootstrap: false})` to not return bootstrap peers.